### PR TITLE
fix: Prevent modal sticky footer from hiding focused elements

### PR DESCRIPTION
**Pull request by jperals**
### Description
A typical problem of sticky headers and footers is that they can cover other elements in an undesired way. In our modal component, the sticky footer can eventually cover focused elements when navigating with the keyboard:

https://user-images.githubusercontent.com/1257272/233566051-15abd309-b2ef-49a2-bae4-6c636de60e1c.mov

This PR fixes this by adding a bottom [scroll padding](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-padding) equivalent to the footer height.

Now the modal scrolls as needed as soon as the focused element meets the footer's position (not the modal's bottom):

https://user-images.githubusercontent.com/1257272/233566062-8c24c29d-5d71-4cee-876e-bce75f8c7e97.mov

### How has this been tested?

- Added integration test
- Manually tested

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/modal/internal.tsx [hunk 1]
```diff
@@ -100,61 +109,68 @@ export default function InternalModal({
...
+      <div
+        {...baseProps}
+        className={clsx(styles.root, { [styles.hidden]: !visible }, baseProps.className, isRefresh && styles.refresh)}
+        role="dialog"
+        aria-modal={true}
+        aria-labelledby={headerId}
+        onMouseDown={onOverlayMouseDown}
+        onClick={onOverlayClick}
+        ref={mergedRef}
+        style={footerHeight ? { scrollPaddingBottom: footerHeight } : undefined}
```
Comment (jperals) [inline src/modal/internal.tsx:127]: Added this `style` prop

File: src/modal/internal.tsx [hunk 2]
```diff
@@ -100,61 +109,68 @@ export default function InternalModal({
...
+                    {header}
+                  </span>
+                </InternalHeader>
+              </div>
+              <div className={clsx(styles.content, { [styles['no-paddings']]: disableContentPaddings })}>
+                {children}
+                <div ref={stickySentinelRef} />
               </div>
+              {footer && (
+                <div ref={footerRef} className={clsx(styles.footer, footerStuck && styles['footer--stuck'])}>
```
Comment (jperals) [inline src/modal/internal.tsx:166]: Added this `ref`

File: pages/modal/scroll-padding.page.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+export default function () {
+  const [visible, setVisible] = useState(false);
+
+  return (
+    <article>
+      <h1>Scroll padding modal</h1>
+      <Button data-testid="modal-trigger" onClick={() => setVisible(true)}>
+        Show modal
+      </Button>
+      <ScreenshotArea>
```
Comment (abdhalees) [inline pages/modal/scroll-padding.page.tsx:16]: why are doing screenshot test on this page?

**Review by abdhalees — DISMISSED** [review]

File: pages/modal/scroll-padding.page.tsx (hunk 3 above)
Comment (jperals) [reply pages/modal/scroll-padding.page.tsx:16]: Not really necessary, just copied over from other pages. I can remove this.

Comment (jperals) [reply pages/modal/scroll-padding.page.tsx]: Done

**Review by abdhalees — APPROVED** [review]

**Pull request merged**